prov/efa/src/rxr/rxr_init.c

```diff
@@ -354,8 +355,21 @@ static int rxr_info_to_rxr(uint32_t version, const struct fi_info *core_info,
 			atomic_ordering = FI_ORDER_ATOMIC_RAR | FI_ORDER_ATOMIC_RAW |
 					  FI_ORDER_ATOMIC_WAR | FI_ORDER_ATOMIC_WAW;
 
-			if (!(hints->tx_attr->msg_order & atomic_ordering))
+			if (hints->tx_attr->msg_order & atomic_ordering) {
+				max_atomic_size = core_info->ep_attr->max_msg_size
```

Comment:
Would this be true even if the atomic is executed with a device read operation?

Reply:
Not sure I understand this question. Would you please elaborate?

Reply:
I see your point, and I assume you meant for fetch atomic. According to the manual:

```
The fetch atomic functions – fi_fetch_atomic, fi_fetch_atomicv, and fi_fetch atomicmsg – behave similar to the equivalent base atomic function. The difference between the fetch and base atomic calls are the fetch atomic routines return the initial value that was stored at the target to the user. The initial value is read into the user provided result buffer. The target buffer of fetch-atomic operations must be enabled for remote read access.
```
If we implement the read of target buffer using device read, the total atomic size would still be bound by the size of base atomic, so this ```max_atomic_size``` calculation still apply.